### PR TITLE
perf: instrumentation improvements

### PR DESCRIPTION
**Pull request by lxsmnsyc**
The goal of this PR is to improve performance of react-scan by:
- reducing callback creation
- reducing optional chaining
- simplifying loops and checks

### CODE REVIEW

File: src/core/instrumentation/index.ts [hunk 1]
```diff
@@ -147,28 +157,37 @@ export const reportRender = (
...
-    if (!fiber.alternate) {
-      return [fiber, null] as const; // use the current fiber as a key
-    }
 
-    const alternateFiberData = ReactScanInternals.reportDataByFiber.get(
-      fiber.alternate,
-    );
-    return [fiber.alternate, alternateFiberData] as const;
-  })();
+function createReportFiber(fiber: Fiber) {
```
Comment (aidenybai) [inline src/core/instrumentation/index.ts:174]: nit: prefer arrow function

File: src/core/instrumentation/index.ts [hunk 2]
```diff
@@ -192,103 +218,20 @@ export const reportRenderFiber = (fiber: Fiber, renders: (Render | null)[]) => {
...
-  onRender,
-  onCommitFinish,
-}: {
-  onCommitStart: () => void;
-  onRender: (fiber: Fiber, render: Render) => void;
-  onCommitFinish: () => void;
-}) => {
-  const handleCommitFiberRoot = (_rendererID: number, root: FiberRoot) => {
+class CommitFiberRootHandler {
+  constructor(public options: InstrumentOptions) {
```
Comment (aidenybai) [inline src/core/instrumentation/index.ts:222]: remove unused constructor?

**Review by aidenybai — COMMENTED** [review]
nits

File: src/core/instrumentation/index.ts (hunk 1 above)
Comment (lxsmnsyc) [reply src/core/instrumentation/index.ts:174]: any particular reason? it's actually shorter (and easier to type with)

```js
function createReportFiber(fiber) {}
// vs
const createReportFiber = (fiber) => {}
```

File: src/core/instrumentation/index.ts (hunk 2 above)
Comment (lxsmnsyc) [reply src/core/instrumentation/index.ts:222]: this isn't unused, it's a shorcut:

```js
constructor(options) {
  this.options = options
}
```

**Comment by lxsmnsyc** [comment]
I'll merge this now and work on offscreen canvas in a new branch.

**Pull request merged**